packages/endo/src/node-module-specifier.js

```diff
@@ -0,0 +1,114 @@
+// q, as in quote, for error messages.
+const q = JSON.stringify;
+
+const isRelative = spec =>
+  spec.startsWith("./") ||
+  spec.startsWith("../") ||
+  spec === "." ||
+  spec === "..";
+
+const normalize = (parts, path) => {
+  for (const part of path) {
```

Comment:
false friend here, I first read `part` as being the loop variable for `parts`. Maybe use `step` instead of `part`?

Reply:
Largely refactored this function to improve clarity.